mirrord-layer/src/socket/hooks.rs

```diff
@@ -24,12 +24,44 @@ pub(crate) unsafe extern "C" fn socket_detour(
     );
 
     let socket_result = FN_SOCKET(domain, type_, protocol);
-    let (Ok(result) | Err(result)) =
-        socket(socket_result, domain, type_, protocol).map_err(From::from);
 
-    trace!("socket_detour -> result {:#?}", result);
+    if IS_INTERNAL_CALL.load(Ordering::Acquire) {
+        debug!("socket_detour -> bypassed");
+        socket_result
+    } else {
+        let (Ok(result) | Err(result)) =
+            socket(socket_result, domain, type_, protocol).map_err(From::from);
+        result
+    }
+}
+
+#[hook_fn]
```

Comment:
I don't think we need this hook?

Reply:
Removed, this was just something that I was investigating.